packages/blocks/src/_common/utils/render-linked-doc.ts

```diff
@@ -346,79 +354,151 @@ async function renderImageAbstract(
   card.isBannerEmpty = false;
 }
 
-function moveBlocksToLinkedDoc(
-  doc: Doc,
-  linkedDoc: Doc,
+export function addBlocksToNote(
+  targetDoc: Doc,
   model: BlockModel,
   parentId: string
 ) {
   // Add current block to linked doc
-  const keys = model.keys as (keyof typeof model)[];
-  const values = keys.map(key => model[key]);
-  const blockProps = Object.fromEntries(keys.map((key, i) => [key, values[i]]));
-  const newModelId = linkedDoc.addBlock(
-    model.flavour as never,
+  const blockProps = getBlockProps(model);
```

Comment:
There might also be some blocks' props happend to be another blocks's id that need to be remapped. Eg. `affine:surface-ref`.

Reply:
Got it. I will open another PR to fix this case specifically.

Reply:
https://github.com/toeverything/blocksuite/assets/12724894/4ba8475f-b0f7-4ec2-b9e5-e16c685bc43c